common/core/nesteddisplay.cpp

```diff
@@ -67,10 +170,110 @@ bool NestedDisplay::SetPowerMode(uint32_t /*power_mode*/) {
   return true;
 }
 
-bool NestedDisplay::Present(std::vector<HwcLayer *> & /*source_layers*/,
+bool NestedDisplay::Present(std::vector<HwcLayer *> &source_layers,
                             int32_t * /*retire_fence*/,
                             bool /*handle_constraints*/) {
-  // TODO implement Present
+#ifdef NESTED_DISPLAY_SUPPORT
+  int ret = 0;
+  size_t size = source_layers.size();
+  const uint32_t* pitches;
+  const uint32_t* offsets;
+  HWCNativeHandle sf_handle;
+  size_t buffer_number = 0;
+  size_t info_size = sizeof(vm_buffer_info);
+  size_t header_size = sizeof(vm_header);
+
+  for (size_t layer_index = 0; layer_index < size; layer_index++) {
+    HwcLayer* layer = source_layers.at(layer_index);
+    if (!layer->IsVisible())
+      continue;
+
+    HwcRect<int> display_frame = layer->GetDisplayFrame();
+    sf_handle = layer->GetNativeHandle();
+    auto search = mHyperDmaExportedBuffers.find(sf_handle);
+    if(search == mHyperDmaExportedBuffers.end()){
+      DrmBuffer* buffer = new DrmBuffer();
```

Comment:
This should be OverlayBuffer::CreateOverlayBuffer();

Reply:
Also, where are u deleting this ?

Reply:
ok, will use exist interface to create drmbuffer. 
I will add the deleting operation in destructor